retrieval/scrape.go

```diff
@@ -531,9 +613,10 @@ func (s samples) Less(i, j int) bool {
 
 func (sl *scrapeLoop) append(b []byte, ts time.Time) (total, added int, err error) {
 	var (
-		app     = sl.appender()
-		p       = textparse.New(b)
-		defTime = timestamp.FromTime(ts)
+		app            = sl.appender()
+		p              = textparse.New(b)
+		defTime        = timestamp.FromTime(ts)
+		samplesScraped = map[string]labels.Labels{}
```

Comment:
This generates garbage on every . Probably fine but we should at least initialize it with the size of the previous one. I'd also rename it to `prevSeries` in the scrapeLoop as it records series not samples.